WordPress/WordPressShareExtension/AppExtensionsService.swift

```diff
@@ -91,9 +91,24 @@ extension AppExtensionsService {
                 onFailure()
                 return
             }
-            let primaryBlogID = ShareExtensionService.retrieveShareExtensionPrimarySite()?.siteID ?? 0
-            let filteredBlogs = blogs.filter({ ($0.blogID.intValue == primaryBlogID || $0.visible == true) })
-            onSuccess(filteredBlogs)
+            // Find the primary site (even if it's not visible)
+            let primarySiteID = ShareExtensionService.retrieveShareExtensionPrimarySite()?.siteID ?? 0
+            let primarySites = blogs.filter({ $0.blogID.intValue == primarySiteID })
+            let visibleSitesSansPrimary = blogs.filter({ $0.blogID.intValue != primarySiteID }).filter({ $0.visible })
+
+            // Recently-used AND visible sites (excluding the primary site)
+            let recentSiteURLs  = ShareExtensionService.retrieveShareExtensionRecentSites() ?? [String()]
+            let recentSites = recentSiteURLs.compactMap({ url in
+                return visibleSitesSansPrimary.first(where: { $0.url == url })
+            })
+
+            // Everything else
+            let remainingSites = visibleSitesSansPrimary.filter({ site in
+                !recentSites.contains(site)
+            })
+
+            let combinedSiteList = primarySites + recentSites + remainingSites
```

Comment:
Would it make sense to encapsulate any of those three blocks in a separate method, so that this line would read similar to:

``
let primarySites = visiblePrimarySites()
let combinedSiteList = primarySites + recentSites(primarySites) + remainingSites(primarySites)
``

Would that make this method easier to understand that way?. If we needed to dig into the details of how primarySites (or recent sites, or remaining sites) are obtained we can dig into the specific method.

Reply:
Makes sense to me @ctarda! I will do that.